rpadmin/admin.go

```diff
@@ -69,6 +69,10 @@ type (
 	NopAuth struct{}
 )
 
+// responseWrapper functions as a testing mechanism for ensuring
```

Comment:
You could also consider adding an `Opt` that allows setting the transport (line 190 in admin.go) and wrap this up in a transport wrapper. Then we'd catch any leaks that might be stemming from pester or a misuse there of.

This alone is a huge improvement though!